client-web/src/exlibris-app/exlibris-book-tile.html

```diff
@@ -0,0 +1,212 @@
+                            src="[[book.image]]">
+                </iron-image>
+                <div id="synopsis">
+                    <div id="title" title="[[book.title]]">[[book.title]]</div>
+                    <div id="author" title="[[book.author]]">[[book.author]]</div>
+                    <div id="status-text" class$="[[statusClass]]">[[book.statusMessage]]</div>
+                </div>
+            </div>
+        </template>
+    </template>
+
+    <script>
+        /**
+         * `<exlibris-book-tile>` components displays short information about book: picture, synopsis and status.
+         *
+         * @customElement
+         * @polymer
+         */
+        class ExlibrisBookTile extends Polymer.Element {
+            static get is() {
+                return 'exlibris-book-tile';
+            }
+
+            static get properties() {
+                return {
+                    /**
+                     * The next objects:
+                     *  @typedef {Object} Book
+                     * @property {string} title the title of a book
+                     * @property {string} authors string with authors of the book
+                     * @property {string} statusMessage full information about status
+                     * @property {string} status  one of the following values: "overdue", "borrowed", "should-return-soon",
+                     * "available", "reserved", "expected","ready-to-pick-up"
+                     */
+                    book: {
+                        type: Object,
+                        observer: '_bookChanged'
+                    },
+                    statusClass: {
+                        value: String,
+                        computed: "_computeClassForStatusMessage(book.status)"
+                    }
+                };
+            }
+
+            /**
+             * Creates statusClass property depending on book.status.
+             *
+             * @param status status of the book.
+             * @returns {string} class for status message.
+             * @private
+             */
+            _computeClassForStatusMessage(status) {
+                if (status === BookStatus.OVERDUE) {
+                    return "warn";
+                }
+                if (status === BookStatus.BORROWED || status === BookStatus.SHOULD_RETURN_SOON) {
+                    return "active";
+                }
+            }
+
+            /**
+             * Creates statusMessage depending on the status
+             *
+             * @param book the book object
+             * @private
+             */
+            _bookChanged(book) {
+                switch (book.status) {
+                    case BookStatus.OVERDUE: {
+                        const oneDay = 1000 * 60 * 60 * 24;
+                        const numberOfOverdueDays = Math.floor((new Date() - book.dueDate) / oneDay);
+                        book.statusMessage = numberOfOverdueDays + " DAYS OVERDUE";
```

Comment:
Why do you change incoming data? `statusMessage` is the computed property of the tile component.